### PR TITLE
Followup on flavor definition in MiniAOD

### PR DESCRIPTION
**Pull request by ahinzmann**
Followup on #10751
Due to comments in #10750

Add jetFlavourInfo to ak4 and ak8 jet collections, to be able to identify gluon splitting to bb.
Add lepton matching for identification of hadronic taus.

With this scheme:
b,c can be identified by either j.hadronFlavour() (hadron definition) or j.partonFlavour() (algorithmic definition) or j.genParton().pdgId() (physics definition)
g,u,d,s can be identified by j.genParton().pdgId() (physics definition)
gluon splitting to bb can be identified roughly by (j.genParton().pdgId()==21 && len(j.jetFlavourInfo().getbHadrons())==2 )

The size increase (edmEventSize) for MiniAOD (on 1300 ttbar events) when adding JetFlavourInfo and lepton matching is the following:
without JetFlavourInfo:
patJets_slimmedJets__PAT. 32977.1 2663.08
patJets_slimmedJetsAK8__PAT. 1141.85 251.157
with JetFlavourInfo:
patJets_slimmedJets__PAT. 33015.2 2696.82
patJets_slimmedJetsAK8__PAT. 1144.26 252.791
with lepton matching:
patJets_slimmedJets__PAT. 33025.3 2707.15
patJets_slimmedJetsAK8__PAT. 1144.78 253.252

### CODE REVIEW

**Comment by monttj** [comment]
please test

**Comment by monttj** [comment]
+1

File: PhysicsTools/JetMCAlgos/python/AK5PFJetsMCFlavourInfos_cfi.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
     jets                     = cms.InputTag("ak5PFJets"),
     bHadrons                 = cms.InputTag("selectedHadronsAndPartons","bHadrons"),
     cHadrons                 = cms.InputTag("selectedHadronsAndPartons","cHadrons"),
-    partons                  = cms.InputTag("selectedHadronsAndPartons","physicsPartons"),
+    partons                  = cms.InputTag("selectedHadronsAndPartons","algorithmicPartons"),
     leptons                  = cms.InputTag("selectedHadronsAndPartons","leptons"),
     jetAlgorithm             = cms.string("AntiKt"),
     rParam                   = cms.double(0.5),
```
Comment (ferencek) [inline PhysicsTools/JetMCAlgos/python/AK5PFJetsMCFlavourInfos_cfi.py]: Can you also set `hadronFlavourHasPriority` to `False` here and in AK4PFJetsMCFlavourInfos_cfi.py and propagate these changes to #11077? Thanks.

**Comment by davidlange6** [comment]
@ahinzmann , all, could this comment from @ferencek be addressed quickly both here and in #11077 so we can finalize the miniaod software.

@arizzi

**Comment by ahinzmann** [comment]
ok working on it.

**Comment by ahinzmann** [comment]
ok, ready to test in 74, 75 and 76.

**Comment by monttj** [comment]
please test

**Pull request merged**